components/chainhook-sdk/src/chainhooks/stacks/mod.rs

```diff
@@ -1325,11 +1327,16 @@ pub fn serialize_stacks_payload_to_json<'a>(
 pub fn handle_stacks_hook_action<'a>(
     trigger: StacksTriggerChainhook<'a>,
     proofs: &HashMap<&'a TransactionIdentifier, String>,
+    config: &PredicatesConfig,
     ctx: &Context,
 ) -> Result<StacksChainhookOccurrence, String> {
     match &trigger.chainhook.action {
         HookAction::HttpPost(http) => {
-            let client = Client::builder()
+            let mut client_builder = Client::builder();
+            if let Some(timeout) = config.payload_http_request_timeout_ms {
+                client_builder = client_builder.timeout(Duration::from_millis(timeout));
+            }
+            let client = client_builder
                 .build()
                 .map_err(|e| format!("unable to build http client: {}", e))?;
```

Comment:
What you think in exchange mutability for method chaining? A mutable `client` instance is less robust than an immutable one, IMO.

And with `unwrap_or_else` a default timeout can be set.

```rust
let client = Client::builder()
    .timeout(config.payload_http_request_timeout_ms
        .map(Duration::from_millis)
        .unwrap_or_else(|| Duration::from_secs(30))) // <<<--- Default timeout
    .build()
    .map_err(|e| format!("unable to build http client: {}", e))?;
```

Reply:
I agree with your suggestion, the problem is that we want the default to be "no timeout" and there's no way to set that via the `ClientBuilder`. I'm not sure why they don't have a method to provide an `Optional<Duration>`

Reply:
Ok. 👍